OMCompiler/Compiler/BackEnd/BackendDAEUtil.mo

```diff
@@ -3074,24 +3093,116 @@ algorithm
+          else isInitial; // Lack of information, consider it just like a normal initial call - warning?
+        end match;
+        if conditionTrue then
+          (_, tpl) := Expression.traverseExpTopDown(expThen, traversingincidenceRowExpSolvableFinder, tpl);
+        else
+          (_, tpl) := Expression.traverseExpTopDown(expElse, traversingincidenceRowExpSolvableFinder, tpl);
+        end if;
+      then tpl;
+    case DAE.IFEXP(expThen = expThen, expElse = expElse)
+      algorithm
+        (_, tpl) := Expression.traverseExpTopDown(expThen, traversingincidenceRowExpSolvableFinder, tpl);
+        (_, tpl) := Expression.traverseExpTopDown(expElse, traversingincidenceRowExpSolvableFinder, tpl);
+      then tpl;
+    else
+      algorithm
+        if Flags.isSet(Flags.FAILTRACE) then
+          Error.addMessage(Error.GENERIC_ELAB_EXPRESSION, {"[BackendDAEUtil.traversingincidenceRowIfExpSolvableFinder]: " + ExpressionDump.dumpExpStr(e, 0) + ": If-Expression could not be evaluated."});
+        end if;
+      then fail();
+  end match;
+end traversingincidenceRowIfExpSolvableFinder;
+
+protected function traversingincidenceRowIfExp
+  "author: kabdelhak FHB 2020-01
+  Checks if condition for special treatment, helper for all traversingincidenceRowExp
+  functions except traversingincidenceRowExpSolvableFinder.
+      special treatment for inital() condition.
+      Ticket: #5795
+      ToDo: inside more complex expression? IF_EQUATION?"
+  input output DAE.Exp e;
+  output Boolean cont = false; // always false, just for convenience
+  input output tuple<BackendDAE.Variables, AvlSetInt.Tree, Boolean> tpl;
+  input traverserFunction traFunc;
+  partial function traverserFunction
+    input output DAE.Exp exp;
+    output Boolean cont;
+    input output tuple<BackendDAE.Variables, AvlSetInt.Tree, Boolean> tpl;
+  end traverserFunction;
+algorithm
+  tpl := match e
+    local
+      DAE.Exp expCond, expThen, expElse;
+      Boolean isInitial, conditionTrue;
+    case DAE.IFEXP(expCond = expCond, expThen = expThen, expElse = expElse) guard(Expression.containsInitialCall(expCond))
+      algorithm
+        (_, _, isInitial) := tpl;
+        conditionTrue := match expCond
+          case DAE.CALL(path=Absyn.IDENT("initial")) then isInitial;
+          case DAE.LUNARY(operator = DAE.NOT(), exp = DAE.CALL(path=Absyn.IDENT("initial"))) then not isInitial;
+          else isInitial; // Lack of information, consider it just like a normal initial call - warning?
+        end match;
+        if conditionTrue then
+          (_, tpl) := Expression.traverseExpTopDown(expThen, traFunc, tpl);
+        else
+          (_, tpl) := Expression.traverseExpTopDown(expElse, traFunc, tpl);
+        end if;
+      then tpl;
+    case DAE.IFEXP(expCond = expCond, expThen = expThen, expElse = expElse)
+      algorithm
+        (_, tpl) := Expression.traverseExpTopDown(expCond, traFunc, tpl);
```

Comment:
Might be worth writing a function that takes a list of condition parts and loops over them

Reply:
We could use the same function for the parts above again